packages/dataviews/src/view-list.tsx

```diff
@@ -21,6 +21,33 @@ import { __ } from '@wordpress/i18n';
  * Internal dependencies
  */
 import { unlock } from './lock-unlock';
+import type {
+	Data,
+	Item,
+	NormalizedField,
+	ViewList as ViewListType,
+} from './types';
+
+type ListViewProps = {
+	view: ViewListType;
+	fields: NormalizedField[];
+	data: Data;
+	isLoading: boolean;
+	getItemId: ( item: Item ) => string;
+	onSelectionChange: ( selection: Item[] ) => void;
+	selection: Item[];
+	id: string;
+};
+
+type ListViewItemProps = {
+	id: string;
+	item: Item;
+	isSelected: boolean;
+	onSelect: ( item: Item ) => void;
+	mediaField?: NormalizedField;
+	primaryField?: NormalizedField;
+	visibleFields: NormalizedField[];
+};
```

Comment:
I'd recommend using `interfaces` in general when we can:

```suggestion
interface ListViewProps {
	view: ViewListType;
	fields: NormalizedField[];
	data: Data;
	isLoading: boolean;
	getItemId: ( item: Item ) => string;
	onSelectionChange: ( selection: Item[] ) => void;
	selection: Item[];
	id: string;
}

interface ListViewItemProps {
	id: string;
	item: Item;
	isSelected: boolean;
	onSelect: ( item: Item ) => void;
	mediaField?: NormalizedField;
	primaryField?: NormalizedField;
	visibleFields: NormalizedField[];
}
```

[There's some rationale here](https://github.com/Automattic/woocommerce-payments/blob/develop/docs/typescript/interface-vs-type.md#should-i-use-interface-or-type-when-declaring-types).

These guidelines in general are pretty good, including the [general conventions](https://github.com/Automattic/woocommerce-payments/blob/develop/docs/typescript/general-conventions.md), I wonder if we should document some guidelines around TypeScript in Gutenberg.

Reply:
We can at least link to them I'd say.